src/color.rs

```diff
@@ -420,6 +420,153 @@ impl Color {
         self.alpha = alpha;
         self
     }
+
+    /// Mix this color with the other given color, with the given weighting .
+    /// 
+    /// The weight determines the amount of the first color that will be used
+    /// in mixing as a percentage. So 50% means use each color equally, while 
+    /// 25% indicates to use one-quarter of this color, and seventy-five percent 
+    /// of the second.
+    /// 
+    /// The alpha channel of both colors are also combined in this way.
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// 
+    /// let c1 = Color::rgba(18., 55., 125., 1.0);
+    /// let mixed = c1.mix(&Color::rgba(125., 33., 200., 0.8), 40);
+    /// assert_eq!(mixed, Color::rgba(72., 44., 163., 0.88));
+    /// ```
+    /// 
+    /// Passing a value for `weight` that is not between 0 and 100 will result
+    /// in `Err(String)` being returned.
+    /// 
+    /// ```should_panic
+    /// use turtle::Color;
+    /// 
+    /// let c1 = Color::rgba(18., 55., 125., 1.0);
+    /// // This will panic
+    /// let mixed = c1.mix(&Color::rgba(125., 33., 200., 0.8), 101);
```

Comment:
Could we use two color strings instead of using `Color::rgba`? You can use the `with_alpha` trick that we did before or maybe use one of the `color::COLOR` constants instead. Also, the comment should mention why the panic would occur.

So something like:
```rust
// This will panic because 101 is not a valid weight. Weights must be between 0 and 100.
```